addons/purchase_stock/models/purchase.py

```diff
@@ -378,6 +378,6 @@ def _create_stock_moves(self, picking):
 
     def _merge_in_existing_line(self, product_id, product_qty, product_uom, location_id, name, origin, values):
         """ This function purpose is to be override with the purpose to forbide _run_buy  method
-        to merge a new po line in an existing one.
+        to merge a new po line in an existing one or to select a specific po line in self.
         """
-        return True
+        return self and self[0] or False
```

Comment:
maybe clairfy the return values?
to me a method named _merge_in_existing_line would return a boolean